src/Perspex.Controls/DockPanel.cs

```diff
@@ -0,0 +1,448 @@
+                    return Alignment.Middle;
+                case VerticalAlignment.Bottom:
+                    return Alignment.End;
+                default:
+                    throw new ArgumentOutOfRangeException(nameof(vert), vert, null);
+            }
+        }
+
+        public static Alignments GetAlignments(this ILayoutable layoutable)
+        {
+            return new Alignments(layoutable.HorizontalAlignment.AsAlignment(), layoutable.VerticalAlignment.AsAlignment());
+        }
+
+        public static Alignments Swap(this Alignments alignments)
+        {
+            return new Alignments(alignments.Vertical, alignments.Horizontal);
+        }
+    }
+
+    public enum Dock
+    {
+        Left = 0,
+        Bottom,
+        Right,
+        Top
+    }
+
+    public static class RectMixin
+    {
+        public static Rect AlignChild(this Rect container, Size childSize, Alignment horizontalAlignment, Alignment verticalAlignment)
+        {
+            var horzSegment = container.GetHorizontalCoordinates();
+            var vertSegment = container.GetVerticalCoordinates();
+
+            var horzResult = GetAlignedSegment(childSize.Width, horizontalAlignment, horzSegment);
+            var vertResult = GetAlignedSegment(childSize.Height, verticalAlignment, vertSegment);
+
+            return FromSegments(horzResult, vertResult);
+        }
+
+        private static Rect FromSegments(Segment horzSegment, Segment vertSegment)
+        {
+            return new Rect(horzSegment.Start, vertSegment.Start, horzSegment.Length, vertSegment.Length);
+        }
+
+        private static Segment GetAlignedSegment(double width, Alignment alignment, Segment horzSegment)
+        {
+            switch (alignment)
+            {
+                case Alignment.Start:
+                    return horzSegment.AlignToStart(width);
+
+                case Alignment.Middle:
+                    return horzSegment.AlignToMiddle(width);
+
+                case Alignment.End:
+                    return horzSegment.AlignToEnd(width);
+
+                default:
+                    return new Segment(horzSegment.Start, horzSegment.End);
+            }
+        }
+
+        private static Segment GetHorizontalCoordinates(this Rect rect)
+        {
+            return new Segment(rect.X, rect.Right);
+        }
+
+        private static Segment GetVerticalCoordinates(this Rect rect)
+        {
+            return new Segment(rect.Y, rect.Bottom);
+        }
+    }
+
+    public struct Segment
+    {
+        public Segment(double start, double end)
+        {
+            Start = start;
+            End = end;
+        }
+
+        public double Start { get; }
+        public double End { get; }
+
+        public double Length => End - Start;
+
+        public override string ToString()
+        {
+            return $"Start: {Start}, End: {End}";
+        }
+    }
+
+    public static class EnumerableMixin
```

Comment:
I'm not sure about fact that you're adding IEnumerable extension methods for everyone who imports Perspex.Controls. Making this internal would at least make them local to just Perspex.